programs/svm-spoke/src/instructions/deposit.rs

```diff
@@ -90,10 +85,7 @@ pub fn deposit_v3(
 
     let current_time = get_current_time(state)?;
 
-    // TODO: if the deposit quote timestamp is bad it is possible to make this error with a subtraction
-    // overflow (from devnet testing). add a test to re-create this and fix it such that the error is thrown,
-    // not caught via overflow.
-    if current_time - quote_timestamp > state.deposit_quote_time_buffer {
+    if current_time.checked_sub(quote_timestamp).unwrap_or(u32::MAX) > state.deposit_quote_time_buffer {
```

Comment:
if `checked_sub` underflows then it defaults to the `unwrap_or` value, of `u32::MAX`, forcing the if to return true, and throwing the error. This fixes the TODO above that I saw on testnet (and validated in unit tests).

Reply:
note that EVM implementation has the same underflow issue, so we might fix it there too

Reply:
in the evm implementation if it fails at that point does it revert with an underflow error or does it throw the associated revert message that we are expecting?

Reply:
as per [this comment](https://github.com/across-protocol/contracts/blob/31693cd43334a2a61916deaf4d0dac2e129d47d2/contracts/SpokePool.sol#L500), it would underflow in EVM

Reply:
right but my question was would it throw the error `InvalidQuoteTimestamp` or `underflow`. it seems it will throw with `underflow (see [here](https://github.com/across-protocol/contracts/blob/chrismaree/deposit-now/test/evm/hardhat/SpokePool.Deposit.ts#L316)).

my original point is that this is actually a bit confusing for a caller. there are situations where multiple invalid input values return the same error. equally, there are situations where some invalid input values, depending on the input, return inconsistent errors.

I agree with your original reply that we should fix this consistently, changing the evm implementation such that all invalid errors return the same error.